src/Model/Timeframe.php

```diff
@@ -282,9 +282,10 @@ public function isValid() {
 				}
 
 				// First we check if the item is already connected to another location to avoid overlapping bookable dates
-				$sameItemTimeframes = \CommonsBooking\Repository\Timeframe::getBookable(
+				$sameItemTimeframes = \CommonsBooking\Repository\Timeframe::get(
 					[],
```

Comment:
Wozu machst du hier die Änderung auf die ::get Funktion? Wenn es um die Sache mit den Restrictions geht, dann solltest du dafür eher einen neuen Parameter in der getBookable ergänzen.

Reply:
Die Idee hinter der Änderung war, wie @chriwen meinte, dass die getBookable Funktion nicht so viele Parameter haben sollte. Wenn jetzt aber die GetBookable Funktion auch prüft, ob der aktuelle Nutzende berechtigt ist den Timeframe zu sehen, dann funktioniert die Funktion nicht bei der Prüfung sich überlappender Timeframes.

Da ja die einzige Funktion der getBookable Funktion vorher war, die get Funktion mit dem zusätzlich fixen Parameter type=BOOKABLE_ID auszuführen, habe ich das einfach auf die zwei Einsatzwecke übertragen, in denen die Funktion nicht für die Frontend-Darstellung genutzt wird.

Ich hab eigentlich nur diese beiden Optionen gesehen:

* Entweder getBookable nimmt mehr Parameter (wie ich es vorher hatte)
ODER
* Die Funktion bekommt einen anderen Zweck und zeigt nur noch für den aktuellen Nutzer buchbare Timeframes an

Die dritte Option wäre vielleicht noch eine getBookable Funktion NUR für die Frontenddarstellung einzuführen, wäre das sinniger?

Danke für euer Feedback! Mir hilft das sehr, mich in den Aufbau von CB reinzudenken!

Reply:
@hansmorb : Auch wenns Christian nicht gefällt, würde ich dich bitten lieber die getBookable mit mehr Parametern zu versehen. Die Variante mit der einer zweiten Funktion z.B. getBookableForCurrentUser fänd ich auch in Ordnung. Die könnte ja einfach die getBookable verwenden und dann über das Ergebnis iterieren wie du es schon vorbereitet hast.